osquery/tables/system/linux/portage.cpp

```diff
@@ -0,0 +1,416 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ *  Portage support by J.O. Aho <trizt@aho.hk>
+ */
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/join.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/trim_all.hpp>
+#include <boost/concept_check.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+#define fs boost::filesystem
+
+namespace osquery {
+namespace tables {
+
+/* File paths used by portage */
+fs::path portagePackageDir = "/var/db/pkg";
+fs::path portageKeywords = "/etc/portage/package.keywords";
+fs::path portageEtcMake = "/etc/make.conf";
+fs::path portageEtcPortageMake = "/etc/portage/make.conf";
+fs::path portageMask = "/etc/portage/package.mask";
+fs::path portageUnMask = "/etc/portage/package.unmask";
+fs::path portageUse = "/etc/portage/package.use";
+fs::path portageWorld = "/var/lib/portage/world";
+
+/* Shared functions and structs */
+
+/**
+ * @brief container to hold data before we store it to QueryData
+ */
+struct portagePackageHolder {
```

Comment:
This needs some work, for classes and structs use upper-camel-case: `PortagePackage`, and we can avoid "Holder" since the first part is pretty clear. :)

For these non-POD structs please inherit from `boost::non_copyable` to help prevent copying all of the complex objects.